### ISSUE TITLE
Data in dispatch closure does not live long enough

### ISSUE THREAD

**Issue by wezm** [Boscop/web-view#25]
Might be related to #15 I built an app with web-view and was experiencing problems with corrupted data and crashes. Using [jemalloc diagnostics](http://jemalloc.net/jemalloc.3.html#mallctl_namespace) I was able to determine that a `String` that was created outside the closure was being dropped, then the closure was using it. Here's how I ran it:

    MALLOC_CONF="junk:true" cargo run

Looking at the code I can see some [lifetime annotations that have been commented out](https://github.com/Boscop/web-view/blob/58d2ec4068f105e19e9274e702a7b4472e1c18e5/src/lib.rs#L113). Uncommenting these immediately gave an error that the value did not live long enough and that I needed to use `move`.

**Comment by Boscop** [comment]
So it works when you write
```rust
pub fn dispatch<F: for<'b> FnMut(&mut WebView<'b, T>, &mut T) + Send + 'a>(&self, f: F) {
```
?

The lifetime `'a` is the lifetime of the underlying webview, but requiring that the closure lives at least as long would be overly restrictive. The closure only has to live until the next time that all queued closures are called (it happens as soon as the thread that started the webview loop is scheduled again). But that's a dynamic lifetime (not known at compile time), so it can't be expressed.
Do you have a better idea for which lifetimes to use here, to avoid overly restricting the lifetime of the closure?

**Comment by zxey** [comment]
Fixed with #36.
